components/Nav/Nav.test.tsx

```diff
@@ -0,0 +1,64 @@
+import { render, screen } from '@testing-library/react';
+import Nav from './Nav';
+
+const mockPush = jest.fn();
+const mockUsePathname = jest.fn();
+
+const mockUseAuthContext = {
+  logoutAccount: jest.fn(),
+};
+
+jest.mock('next/navigation', () => ({
+  useRouter() {
+    return {
+      push: mockPush,
+    };
+  },
+  usePathname() {
+    return mockUsePathname();
+  },
+}));
+
+jest.mock('../../context/AuthContextProvider', () => ({
+  useAuthContext() {
+    return {
+      ...mockUseAuthContext,
+    };
+  },
+}));
+
+describe('Nav', () => {
+  beforeAll(() => {
+    jest.clearAllMocks();
+  });
+  it('it should be hidden when path is /register', () => {
+    mockUsePathname.mockImplementation(() => '/register');
+
+    render(<Nav />);
```

Comment:
Q: Are we able to render `<Nav/>` in the `beforeAll` method since we're rendering it for every test?

Reply:
Sure can; I like this idea.

Reply:
So I did try putting the render `<Nav/>` in the `beforeAll` method after seeing @ryandotfurrer tests. However, the tests that had a route change (/login and /register) failed because the mock routes are being set after the render. Only the test that didn't have the route change passed so I moved the render `<Nav/>` back into each test.

Reply:
If you need to render a component after a state change, beforeEach, beforeAll, etc, won't work. It will render before the state changes.